data-upgrade-users/src/main/java/org/exoplatform/migration/UsersLastLoginTimeMigration.java

```diff
@@ -0,0 +1,111 @@
+import org.exoplatform.commons.utils.CommonsUtils;
+import org.exoplatform.commons.utils.ListAccess;
+import org.exoplatform.container.xml.InitParams;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+import org.exoplatform.services.organization.OrganizationService;
+import org.exoplatform.services.organization.User;
+import org.exoplatform.social.core.identity.model.Identity;
+import org.exoplatform.social.core.identity.model.Profile;
+import org.exoplatform.social.core.identity.provider.OrganizationIdentityProvider;
+import org.exoplatform.social.core.jpa.search.ProfileIndexingServiceConnector;
+import org.exoplatform.social.core.manager.IdentityManager;
+import org.exoplatform.social.core.profile.ProfileFilter;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+public class UsersLastLoginTimeMigration extends UpgradeProductPlugin {
+  private static final Log    LOG = ExoLogger.getExoLogger(UsersLastLoginTimeMigration.class);
+
+  private OrganizationService organizationService;
+
+  private IdentityManager     identityManager;
+
+  public UsersLastLoginTimeMigration(OrganizationService organizationService,
+                                     IdentityManager identityManager,
+                                     InitParams initParams) {
+    super(initParams);
+    this.organizationService = organizationService;
+    this.identityManager = identityManager;
+  }
+
+  @Override
+  public void processUpgrade(String oldVersion, String newVersion) {
+    LOG.info("Start upgrade process to add lastLoginTime in user profile");
+    long startupTime = System.currentTimeMillis();
+    int limit = 100;
+    int offset = 0;
+    int totalSize = 0;
+    ListAccess<Identity> listIdentity = null;
+    try {
+      ProfileFilter filter = new ProfileFilter();
+      filter.setConnected(false);
+      listIdentity = identityManager.getIdentitiesByProfileFilter(OrganizationIdentityProvider.NAME, filter, true);
+      totalSize = listIdentity.getSize();
+      LOG.info("Number of users to check : " + totalSize);
+      int limitToFetch = limit;
+      List<Identity> identities;
+      do {
+        if (totalSize < (offset + limitToFetch)) {
+          limitToFetch = totalSize - offset;
+        }
+        identities = Arrays.asList(listIdentity.load(offset, limitToFetch));
+        int itemsMigratedPerLoop = updateLastLoginTime(identities);
+        offset = (offset + limitToFetch) - itemsMigratedPerLoop;
+      } while (offset < totalSize);
+    } catch (Exception e) {
+      LOG.error("Error processUpgrade data-upgrade-users", e);
+    }
+    LOG.info("End process to add lastLoginTime in user profile. It took {} ms all migration {}",
+             (System.currentTimeMillis() - startupTime));
+  }
+
+  public int updateLastLoginTime(List<Identity> identities) throws Exception {
+    int totalSize = identities.size();
+    int totalItemsChecked = 0;
```

Comment:
this parameter should be also passed in parameter.
This number should represent how many users treaten since the beginning of the migration, not from the beginning of the current bunch